testbed/eks/lib/addons/fluxv2.ts

```diff
@@ -0,0 +1,110 @@
+import cdk = require('@aws-cdk/core');
+import eks = require('@aws-cdk/aws-eks');
+import * as yaml from 'js-yaml';
+import * as request from 'sync-request';
+
+export interface FluxV2Props extends cdk.StackProps {
+    cluster: eks.Cluster;
+    fluxRepos: Repository[];
+    secretName: string;
+    fluxVersion: string;
+}
+export class Repository {
+    repoUrl: string;
+    repoBranch: string;
+    repoPath: string;
+
+    constructor(repoUrl: string, repoBranch?: string, repoPath?: string) {
+        if (!repoBranch) {
```

Comment:
Check out typescript's defaulting syntax
`this.branch = this.branch ?? "main"`

https://stackoverflow.com/questions/47328862/how-to-set-a-variable-if-undefined-in-typescript

Reply:
oh man! I need to learn typescript!

Reply:
haha! I'm new to typescript as well. This is my first time writing typescript lol